api/argo-templates/ncn/worker.drain.yaml

```diff
@@ -22,36 +22,108 @@
 # OTHER DEALINGS IN THE SOFTWARE.
 #
 {{define "worker.drain"}}
-apiVersion: argoproj.io/v1alpha1
-kind: Workflow
-metadata:
-  generateName: sub-workflow-drain-{{"{{inputs.parameters.targetNcn}}"}}-
-spec:
-  podGC:
-    # Pod GC strategy must be one of the following:
-    # * OnPodCompletion - delete pods immediately when pod is completed (including errors/failures)
-    # * OnPodSuccess - delete pods immediately when pod is successful
-    # * OnWorkflowCompletion - delete pods when workflow is completed
-    # * OnWorkflowSuccess - delete pods when workflow is successful
-    strategy: OnPodCompletion
-  affinity:
-    nodeAffinity:
-      requiredDuringSchedulingIgnoredDuringExecution:
-        nodeSelectorTerms:
-        - matchExpressions:
-          - key: kubernetes.io/hostname
-            operator: NotIn
-            values:
-            {{- range $index,$value := .TargetNcns }}
-            - {{$value -}}
-            {{- end }}
-  arguments:
-    parameters:
-    - name: scriptContent
-      value: |
-        sleep $((RANDOM % 10))
-    - name: dryRun
-      value: {{"{{inputs.parameters.dryRun}}"}}
-  workflowTemplateRef:
-    name: {{"{{inputs.parameters.workflowtemplate}}"}}
+tasks:
+  - name: wait-for-cfs
+    templateRef:
+      name: ssh-with-kubeconfig
+      template: bash-script-example
+    arguments:
+      parameters:
+        - name: dryRun
+          value: "{{ `{{inputs.parameters.dryRun}}` }}"
+        - name: scriptContent
+          value: |
+            /usr/share/doc/csm/upgrade/1.2/scripts/cfs/wait_for_configuration.sh \
+              --xnames \
+              $(ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null {{ `{{inputs.parameters.targetNcn}}` }} 'cat /etc/cray/xname')
+  - name: ensure-nexus-can-start-on-any-node
+    templateRef:
+      name: ssh-with-kubeconfig
+      template: bash-script-example
+    arguments:
+      parameters:
+        - name: dryRun
+          value: "{{ `{{inputs.parameters.dryRun}}` }}"
+        - name: scriptContent
+          value: |
+            workers="$(kubectl get node --selector='!node-role.kubernetes.io/master' -o name | sed -e 's,^node/,,' | paste -sd,)"
+            export PDSH_SSH_ARGS_APPEND="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+            kubectl get configmap -n nexus cray-precache-images -o json | jq -r '.data.images_to_cache' | while read image; do echo >&2 "+ caching $image"; pdsh -w "$workers" "crictl pull $image" 2>/dev/null; done
```

Comment:
Should be able to do something like (untested) and save the jq, not sure if .data.images_to_cache is an array of data or not if not can use nuke the range.

```
kubectl get pods -o=jsonpath='{range .data.images_to_cache}{"/n"}' 
```